csrc/codegen.cpp

```diff
@@ -1584,17 +1584,17 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     const auto data_type = grop->out()->dtype();
     const auto op_type = grop->getReductionOpType();
 
+    if (grop->isSerial()) {
+      generateSerialGridReduction(grop);
+      return;
+    }
+
```

Comment:
This block is moved earlier to avoid dereferencing null pointers for reduction and sync buffer, which are not set for serial reductions.